packages/application-generic/src/usecases/map-trigger-recipients/map-trigger-recipients.use-case.ts

```diff
@@ -159,8 +165,33 @@ export class MapTriggerRecipients {
     return subscriber;
   }
 
-  private findSubscribers(recipients: TriggerRecipients): ISubscribersDefine[] {
-    return recipients.filter(isNotTopic).map(this.mapSubscriber);
+  public mapSubscriber(
+    subscriber: TriggerRecipientSubscriber,
+    source: SubscriberSourceEnum
+  ): ISubscribersSource | null {
+    if (!subscriber) return null;
+
+    let mappedSubscriber: Partial<ISubscribersSource>;
+
+    if (typeof subscriber === 'string') {
+      mappedSubscriber = { subscriberId: subscriber };
```

Comment:
This runtime type check here is a little uneasy. We should really only have a single object type.

Is this due to legacy tech debt? Is it easy to fix?

Reply:
Correct this is a legacy logic in order to align the subscriber interface.
Please note that at this addition code, we did not add any additional iterations/checks.
We slitted `mapSubscriber` to two functions of `mapActor` and `mapSubscriber` so that for `mapActor` we do not add `_subscriberSource` because it does not make sense, and for `mapSubscriber` we add `_subscriberSource`.